Client/U2FExtensions.swift

```diff
@@ -0,0 +1,893 @@
+                guard let handle = fido2RegHandles.first else {
+                    log.error(U2FErrorMessages.ErrorRegistration)
+                    return
+                }
+                guard let request = fido2RegisterRequest[handle] else {
+                    sendFIDO2RegistrationError(handle: handle)
+                    return
+                }
+                handleFIDO2Registration(handle: handle, request: request)
+            }
+            
+            if !fidoRegHandles.isEmpty {
+                guard let handle = fidoRegHandles.first else {
+                    log.error(U2FErrorMessages.ErrorRegistration)
+                    return
+                }
+                guard let request = fidoRegisterRequest[handle] else {
+                    sendFIDORegistrationError(handle: handle, requestId: requestId[handle] ?? -1, errorCode: U2FErrorCodes.other_error)
+                    return
+                }
+                handleFIDORegistration(handle: handle, request: request, requestId: requestId[handle] ?? -1)
+            }
+        
+            if !fido2AuthHandles.isEmpty {
+                guard let handle = fido2AuthHandles.first else {
+                    log.error(U2FErrorMessages.ErrorAuthentication)
+                    return
+                }
+                guard let request = fido2AuthRequest[handle] else {
+                    sendFIDO2AuthenticationError(handle: handle)
+                    return
+                }
+                handleFIDO2Authentication(handle: handle, request: request)
+            }
+        
+            if !fidoSignHandles.isEmpty {
+                guard let handle = fidoSignHandles.first else {
+                    log.error(U2FErrorMessages.ErrorAuthentication)
+                    return
+                }
+                guard let keys = fidoSignRequests[handle] else {
+                    sendFIDOAuthenticationError(handle: handle, requestId: requestId[handle] ?? -1, errorCode: U2FErrorCodes.other_error)
+                    return
+                }
+                handleFIDOAuthentication(handle: handle, keys: keys, requestId: requestId[handle] ?? -1)
+            }
+        }
+    }
+}
+
+// MARK: - TabContentScript
+extension U2FExtensions: TabContentScript {
+    static func name() -> String {
+        return "U2F"
+    }
+    
+    func scriptMessageHandlerName() -> String? {
+        return "U2F"
+    }
+    
+    func userContentController(_ userContentController: WKUserContentController, didReceiveScriptMessage message: WKScriptMessage) {
+        if message.name == "U2F", let body = message.body as? NSDictionary {
+            
+            guard let name = body["name"] as? String, let handle = body["handle"] as? Int else {
+                log.error(U2FErrorMessages.Error)
+                return
+            }
+            
+            // FIDO2 is the new webauthn API
+            // https://www.w3.org/TR/webauthn/
+            if name == "fido2-create" {
```

Comment:
Might be nice to throw these names into an enum. Either way, we should probably be using a switch statement since we're not going to get more than 1 of these flows at once